File: recursion/reverse_string.js
var reverseString = (s) => {
  function reverse(i, j) {
    if (i >= j) return;
    [s[i], s[j]] = [s[j], s[i]];
    reverse(i + 1, j - 1);
  }

  reverse(0, s.length - 1);
};

File: general/arrays_and_hashing/reverse_string.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        return self.helper(0, len(s) - 1, s)

    def helper(self, left, right, s):
        # Base case
        if left >= right:
            return

        # General case
        s[left], s[right] = s[right], s[left]

        self.helper(left + 1, right - 1, s)

# Time Complexity: O(n)
# Space Complexity: O(1)